src/SimpleSAML/Metadata/Sources/MDQ.php

```diff
@@ -183,7 +183,8 @@ private function getFromCache(string $set, string $entityId): ?array
             ));
         }
 
-        $data = json_decode($rawData);
+        // ensure json is decoded as an associative array not an object
+        $data = json_decode($rawData, true);
         if ($data === false) {
```

Comment:
Maybe replace this block with the `JSON_THROW_ON_ERROR` flag added to json_decode.

We could consider adding our own json util function that does 'correctly' process any json and handle errors as we use json_decode in several places around the code. This is not in scope for this PR though, it can be merged as it is now..